cassiopeia/simulator/BirthDeathFitnessSimulator.py

```diff
@@ -0,0 +1,399 @@
+                tree.nodes[unique_id]["time"] = self.experiment_time
+
+                current_lineages.put(
+                    (
+                        self.experiment_time,
+                        unique_id,
+                        {
+                            "id": unique_id,
+                            "birth_scale": lineage["birth_scale"],
+                            "total_time": self.experiment_time,
+                            "active": False,
+                        },
+                    )
+                )
+                observed_nodes.append(unique_id)
+
+            else:
+                if birth_waiting_time < death_waiting_time:
+                    # Update birth rate
+                    updated_birth_scale = update_birth_scale(
+                        lineage["birth_scale"]
+                    )
+
+                    # Annotate parameters for a given node in the tree
+                    tree.add_node(unique_id)
+                    tree.nodes[unique_id]["birth_scale"] = updated_birth_scale
+                    tree.add_edge(lineage["id"], unique_id)
+                    tree.nodes[unique_id]["time"] = (
+                        birth_waiting_time + lineage["total_time"]
+                    )
+                    # Add the newly generated cell to the list of living lineages
+                    current_lineages.put(
+                        (
+                            birth_waiting_time + lineage["total_time"],
+                            unique_id,
+                            {
+                                "id": unique_id,
+                                "birth_scale": updated_birth_scale,
+                                "total_time": birth_waiting_time
+                                + lineage["total_time"],
+                                "active": True,
+                            },
+                        )
+                    )
+
+                else:
+                    tree.add_node(unique_id)
+                    tree.nodes[unique_id]["birth_scale"] = lineage[
+                        "birth_scale"
+                    ]
+                    tree.add_edge(lineage["id"], unique_id)
+                    tree.nodes[unique_id]["time"] = (
+                        death_waiting_time + lineage["total_time"]
+                    )
+                    current_lineages.put(
+                        (
+                            death_waiting_time + lineage["total_time"],
+                            unique_id,
+                            {
+                                "id": unique_id,
+                                "birth_scale": lineage["birth_scale"],
+                                "total_time": death_waiting_time
+                                + lineage["total_time"],
+                                "active": False,
+                            },
+                        )
+                    )
+
+        def update_birth_scale(birth_scale):
```

Comment:
Shouldn't this function take in `self`?

Reply:
No I don't think so, as it's defined in the body of `simulate tree`

Reply:
Oh I didn't notice! Thanks Richard!